src/horizontal-pod-autoscaler.ts

```diff
@@ -0,0 +1,866 @@
+   */
+  readonly scaleUp?: ScalingRules;
+  /**
+   * The scaling behavior when scaling down.
+   *
+   * @default
+   * Scale down to minReplica count with a 5 minute stabilization window.
+   */
+  readonly scaleDown?: ScalingRules;
+}
+
+/**
+ * A HorizontalPodAutoscaler scales a workload up or down in response to a metric
+ * change. This allows your services to scale up when demand is high and scale down
+ * when they are no longer needed.
+ *
+ *
+ * The following are typical use cases for HorizontalPodAutoscaler:
+ *
+ * * When Memory usage is above 70%, scale up the number of replicas to meet the demand.
+ * * When CPU usage is below 30%, scale down the number of replicas to save resources.
+ * * When a service is experiencing a spike in traffic, scale up the number of replicas
+ *   to meet the demand. Then, when the traffic subsides, scale down the number of
+ *   replicas to save resources.
+ *
+ * HorizontalPodAutoscaler's can be used to scale the following workloads:
+ * * Deployment
+ * * ReplicaSet
+ * * ReplicationController
+ * * StatefulSet
+ *
+ * :warning: HorizontalPodAutoscaler cannot be used with a DaemonSet.
+ *
+ * ### Targets that already have a replica count defined.
+ *
+ * It is recommended to remove any replica counts from the target resource before using a
+ * HorizontalPodAutoscaler. If this isn't done, then any time a change to that object is applied,
+ * Kubernetes will scale the current number of Pods to the value of the target.replicas key. This
+ * may not be desired and could lead to unexpected behavior.
+ *
+ *
+ * @see https://kubernetes.io/docs/tasks/run-application/horizontal-pod-autoscale/#implicit-maintenance-mode-deactivation
+ *
+ * @example
+ * ```ts
+ * const backend = new kplus.Deployment(this, 'Backend', ...);
+ *
+ * const hpa = new kplus.HorizontalPodAutoscaler(chart, 'Hpa', {
+ *  target: backend,
+ *  maxReplicas: 10,
+ *  scaleUp: {
+ *    policies: [
+ *      {
+ *        replicas: kplus.Replicas.absolute(3),
+ *        duration: Duration.minutes(5),
+ *      },
+ *    ],
+ *  },
+ * });
+ * ```
+ */
+export class HorizontalPodAutoscaler extends Resource {
+  /**
+ * @see base.Resource.apiObject
+ */
+  protected readonly apiObject: ApiObject;
+  public readonly resourceType = 'horizontalpodautoscaler';
+  /**
+   * The workload to scale up or down.
+   *
+   * Scalable workload types:
+   * * Deployment
+   * * ReplicaSet
+   * * ReplicationController
+   * * StatefulSet
+   *
+   * Cannot be used with workloads that do not scale, such as DaemonSet.
+   */
+  public readonly target: Workload;
+  /**
+   * The maximum number of replicas that can be scaled up to.
+   */
+  public readonly maxReplicas: number;
+  /**
+   * The minimum number of replicas that can be scaled down to.
+   *
+   * Can be set to 0 if the alpha feature gate `HPAScaleToZero` is enabled and
+   * at least one Object or External metric is configured.
+   *
+   * @default 1
```

Comment:
Again, output properties should speak to what is currently defined on the instance, not what are the possible configurations. 

```suggestion
   * The minimum number of replicas that can be scaled down to.
```

This repeats throughout the docs of output properties, please make sure to go through all of them.